packages/shared/src/components/HorizontalScroll/HorizontalScroll.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  RefObject,
+  MouseEventHandler,
+} from 'react';
+import classNames from 'classnames';
+import { useHorizontalScrollHeader } from './useHorizontalScrollHeader';
+
+interface HorizontalScrollProps {
+  title: string | ReactElement;
+  children: ReactNode;
+  onScroll?: (ref: RefObject<HTMLElement>) => void;
+  onClickSeeAll?: MouseEventHandler;
+  className?: string;
+}
+
+export default function HorizontalScroll({
+  title,
+  children,
+  onScroll,
+  onClickSeeAll,
+  className,
+}: HorizontalScrollProps): ReactElement {
+  const { ref, Header } = useHorizontalScrollHeader({
+    title,
+    onScroll,
+    onClickSeeAll,
+  });
+
+  return (
+    <>
+      <Header />
+      <div
+        ref={ref}
+        className={classNames(
+          'no-scrollbar grid auto-cols-max grid-flow-col overflow-x-scroll scroll-smooth',
+          className,
+        )}
+        role="region"
+        aria-label="Horizontal Scroll"
```

Comment:
If we have always the `<Header />` component with the title for this section maybe we would like to use `aria-labelledby` pointing to the title. 
So user that enter into this section with a screen reader will hear something like: `Related posts` instead `Horizontal Scroll`. What do you think?

Reference: https://www.aditus.io/aria/aria-label/#aria-label-vs-aria-labelledby

Reply:
Thank u!! Will change :D